documentation/add/07-sqw_redesign.md

```diff
@@ -1,58 +1,84 @@
+
+Extraction from the whole inheritance diagram for the `DnD` objects is presented on the Fig.3:
+
+![Fig.3. DnD Class inheritance diagram](../diagrams/DND-inheritance.png)
+
+The diagram also shows `data_sqw_dnd` object inheriting from `DnDBase` and containing arbitrary number of dimensions unlike other `DnD` objects which define their specific number of dimensions.  This object left for IO compatibility with previous versions of Horace code and is not used in Horace for any other purpose except IO restoring old data.
+
+The `DnDBase` base class is an abstract class holding the common data and common code, including the operation manager which is responsible for matching dimensions between the specific `DnD` objects before executing appropriate algorithms. The structure of `DnDBase` class together with its children, defining specific number of dimensions is presented on Fig.4:
+
+![Fig.4. DND Class Overview](../diagrams/dnd.png)
+
+The diagram shows that any `DnD` object contains *signal*, *error* and *npix* (*npix* describes the number of pixels (neutron events) contributed to the appropriate area of the image and related to the image scaling) arrays of appropriate dimensions and two additional properties, namely *proj* and *axes*. *Proj* field contains the instance of `aProjection` class, which defines the transformation from the coordinate system of `PixelData` class to the `Image` (`DnD` object) coordinate system. For example, in the most common case of `Image` coordinate system being `hkl-dE` coordinate system and `PixelData` class coordinate system being Crystal Cartesian coordinate system the transformation is the matrix multiplication of the `PixelData` coordinates by `UB` matrix of Bussing and Levy [^1].
+
+The `axes_block` class is closely related to the appropriate `aProjection` class and defines the coordinate system of the image, its binning axes and units along the axes scales.
+
+In more details the `aProjection` class and `axes_block` classes are described below.
 
-The `DnDBase` base class is an abstract class holding the common code, including the operation manager which is responsible for matching dimensions between the specific `DnD` objects before executing.
+### Plotting interface
+
+The 1D-3D`sqw` and `dnd` Horace objects can be plotted with the appropriate dimensions plots. There are large number of plotting functions so it is reasonable to extract these functions into separate plotting interface. `sqw`, `dnd` and Herbert `IX_dataset` object inherit `data_plot_interface` and implement particular plotting functions specific to particular n-dimensional objects. The inheritance diagram of this plotting interface is presented on the Fig.5:
```

Comment:
add A before large number and another A before separate.
presented ON->presented IN 
remove the before Fig